radix-transaction-scenarios/src/scenarios/all_scenarios.rs

```diff
@@ -1,54 +1,422 @@
+                                &receipt,
+                                &self.database,
+                            );
+                            previous = Some(receipt);
+                        }
+                        NextAction::Completed(end_state) => {
+                            match self.nonce_handling {
+                                NonceHandling::Increment(increment) => next_nonce += increment,
+                                NonceHandling::StartAtNextAvailable => {
+                                    next_nonce = end_state.next_unused_nonce
+                                }
+                            }
+                            break;
+                        }
+                    }
+                }
             }
-            7 => Some(Box::new(|core| {
-                global_n_owned::GlobalNOwnedScenarioCreator::create(core)
-            })),
-            8 => Some(Box::new(|core| {
-                non_fungible_resource_with_remote_type::NonFungibleResourceWithRemoteTypeScenarioCreator::create(core)
-            })),
-            9 => Some(Box::new(|core| {
-                kv_store_with_remote_type::KVStoreScenarioCreator::create(core)
-            })),
-            10 => Some(Box::new(|core| {
-                max_transaction::MaxTransactionScenarioCreator::create(core)
-            })),
-            11 => Some(Box::new(|core| {
-                royalties::RoyaltiesScenarioCreator::create(core)
-            })),
-            _ => None,
         }
+
+        Ok(CompletedExecutionReceipt {
+            database: self.database,
+        })
+    }
+
+    fn execute_transaction(
+        &mut self,
+        transaction: &RawNotarizedTransaction,
+    ) -> Result<TransactionReceiptV1, ScenarioExecutorError> {
+        let network = NetworkDefinition::simulator();
+        let validator = NotarizedTransactionValidator::new(ValidationConfig::default(network.id));
+        let validated = validator
+            .validate_from_raw(transaction)
+            .map_err(ScenarioExecutorError::TransactionValidationError)?;
+
+        let receipt = execute_transaction(
+            &self.database,
+            Vm::new(&self.scrypto_vm, self.native_vm.clone()),
+            &CostingParameters::default(),
+            &ExecutionConfig::for_notarized_transaction(self.network_definition.clone()),
+            &validated.get_executable(),
+        );
+
+        if let TransactionResult::Commit(commit) = &receipt.result {
+            let database_updates = commit
+                .state_updates
+                .create_database_updates::<SpreadPrefixKeyMapper>();
+            self.database.commit(&database_updates);
+        };
+
+        Ok(receipt)
     }
 }
+
+#[derive(Clone, Debug)]
+pub enum ScenarioExecutorError {
+    BootstrapFailed,
+    FailedToGetEpoch,
+    TransactionValidationError(TransactionValidationError),
+}
+
+#[derive(Clone, Debug)]
+pub enum NonceHandling {
+    Increment(u32),
```

Comment:
Perhaps a clearer name might be `ScenarioStartNonceHandling`, and then the `Increment` variant might be more clearly named `PreviousScenarioStartNoncePlus(u32)` and `StartAtNextAvailable` could be `PreviousScenarioEndNoncePlusOne`

Reply:
Sounds good, I've made those changes.